dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Api::V1::SectionsController < ApplicationController
+  layout false
+
+  # Don't bother redirecting to login when denying access to the JSON APIs
+  rescue_from CanCan::AccessDenied do
+    head :forbidden
+  end
+
+  # GET /api/v1/sections
+  # Get the set of sections owned by the current user
+  def index
+    return head :forbidden unless current_user
+    render json: current_user.sections.map(&:summarize)
+  end
+
+  # GET /api/v1/sections/<id>
+  # Get complete details of a particular section
+  def show
+    section = Section.find(params[:id])
+    authorize! :read, section
+    render json: section.summarize
+  end
```

Comment:
Note, this will be a slight behavior change in that students (who are not authorized to view sections) will get `forbidden` for the index instead of an empty set as they do now. I think that may be desirable.